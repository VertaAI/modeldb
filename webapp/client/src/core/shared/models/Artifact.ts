import { JsonObject, JsonProperty } from 'json2typescript';

export interface IArtifact {
  key: string;
  path: string;
  type: ArtifactType;
  pathOnly: boolean;
  linkedArtifactId?: string;
  fileExtension?: string;
}
export interface IArtifactWithPath extends IArtifact {
  pathOnly: false;
}

export const checkArtifactWithPath = (
  artifact: IArtifact
): artifact is IArtifactWithPath => {
  return Boolean(artifact.path && !artifact.pathOnly);
};

@JsonObject('artifact')
export class Artifact implements IArtifact {
  @JsonProperty('key', String)
  public readonly key: string;
  @JsonProperty('path', String)
  public readonly path: string;
  @JsonProperty('artifact_type', String, true)
  public readonly type: ArtifactType;
  @JsonProperty('path_only', Boolean, true)
  public readonly pathOnly: boolean;
  @JsonProperty('linked_artifact_id', String, true)
  public linkedArtifactId?: string = '';
  @JsonProperty('filename_extension', String, true)
  public fileExtension?: string = undefined;

  public constructor(
    pathOnly: boolean,
    type: ArtifactType,
    key: string,
    path?: string
  ) {
    this.key = key || '';
    this.path = path || '';
    this.type = type || 'BLOB';
    this.pathOnly = pathOnly || false;
  }
}

export type ArtifactType =
  | 'IMAGE'
  | 'BLOB'
  | 'BINARY'
  | 'QUERY'
  | 'DATA'
  | 'CODE';
