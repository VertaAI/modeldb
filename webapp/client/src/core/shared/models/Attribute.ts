import { JsonObject, JsonProperty } from 'json2typescript';

import { AnyToStringOrNumberConverter } from 'core/shared/utils/mapperConverters';

import { IKeyValuePair } from './Common';

export type IAttribute = IKeyValuePair<
  string | number | Array<string | number>
>;

@JsonObject('Attribute')
export class Attribute implements IAttribute {
  @JsonProperty('key', String, true)
  public readonly key: string;
  @JsonProperty('value', AnyToStringOrNumberConverter)
  public readonly value: IAttribute['value'];

  constructor(key?: string, value?: IAttribute['value']) {
    this.key = key || '';
    this.value = value || '' || 0;
  }
}
