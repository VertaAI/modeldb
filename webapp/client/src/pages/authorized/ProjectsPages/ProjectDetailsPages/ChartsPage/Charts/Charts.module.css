.root {
  display: flex;
  flex-direction: column;
  width: 100%;
}
