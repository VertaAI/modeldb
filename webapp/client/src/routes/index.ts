import makeRoute, { IRoute as _IRoute } from 'core/shared/routes/makeRoute';
import { ArgumentTypes } from 'core/shared/utils/types';

import { makeRouteWithWorkspace } from './routeWithWorkspace';

export type IRoute<T, B = undefined> = _IRoute<T, B>;

export type RoutesWithWorkspaces = 'projects' | 'datasets';

const routes = {
  index: makeRoute({
    getPath: () => '/',
  }),

  workspace: makeRouteWithWorkspace({
    getPath: () => '/',
  }),

  datasets: makeRouteWithWorkspace({
    getPath: () => '/datasets',
  }),
  datasetCreation: makeRouteWithWorkspace({
    getPath: () => '/datasets/new',
  }),
  datasetSummary: makeRouteWithWorkspace<{
    datasetId: string;
  }>({
    getPath: () => '/datasets/:datasetId/summary',
  }),
  datasetVersions: makeRouteWithWorkspace<{
    datasetId: string;
  }>({
    getPath: () => '/datasets/:datasetId/versions',
  }),
  datasetVersion: makeRouteWithWorkspace<{
    datasetId: string;
    datasetVersionId: string;
  }>({
    getPath: () => '/datasets/:datasetId/versions/:datasetVersionId',
  }),
  compareDatasetVersions: makeRouteWithWorkspace<{
    datasetId: string;
    datasetVersionId1: string;
    datasetVersionId2: string;
  }>({
    getPath: () =>
      '/datasets/:datasetId/versions/compare/:datasetVersionId1/:datasetVersionId2',
  }),

  projects: makeRouteWithWorkspace({
    getPath: () => `/projects`,
  }),
  projectCreation: makeRouteWithWorkspace({
    getPath: () => '/projects/new',
  }),
  projectSummary: makeRouteWithWorkspace<{
    projectId: string;
  }>({
    getPath: () => `/projects/:projectId/summary`,
  }),
  experimentRuns: makeRouteWithWorkspace<{
    projectId: string;
  }>({
    getPath: () => '/projects/:projectId/exp-runs',
  }),
  charts: makeRouteWithWorkspace<{
    projectId: string;
  }>({
    getPath: () => '/projects/:projectId/charts',
  }),
  experiments: makeRouteWithWorkspace<{
    projectId: string;
  }>({
    getPath: () => '/projects/:projectId/experiments',
  }),
  experimentCreation: makeRouteWithWorkspace<{
    projectId: string;
  }>({
    getPath: () => '/projects/:projectId/experiments/new',
  }),
  modelRecord: makeRouteWithWorkspace<{
    projectId: string;
    modelRecordId: string;
  }>({
    getPath: () => '/projects/:projectId/exp-runs/:modelRecordId',
  }),
  compareModels: makeRouteWithWorkspace<{
    projectId: string;
    modelRecordId1: string;
    modelRecordId2: string;
  }>({
    getPath: () =>
      '/projects/:projectId/exp-runs/compare/:modelRecordId1/:modelRecordId2',
  }),
};

export type GetRouteParams<T extends IRoute<any, any>> = ArgumentTypes<
  T['getRedirectPath']
>[0];
export type GetRouteQueryParams<
  T extends _IRoute<any, any>
> = T extends _IRoute<any, infer QueryParams> ? QueryParams : {};
export default routes;
