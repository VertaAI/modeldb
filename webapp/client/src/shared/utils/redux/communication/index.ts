export {
  default as makeCommunicationActionTypes,
} from './makeCommunicationActionTypes';
export {
  default as makeCommunicationReducer,
} from './makeCommunicationReducer';
export {
  default as makeCommunicationReducerFromEnum,
} from './makeCommunicationReducerFromEnum';
export {
  default as makeCommunicationReducerByIdFromEnum,
} from './makeCommunicationReducerByIdFromEnum';
export * from './types';
export {
  initialCommunication,
  successfullCommunication,
  requestingCommunication,
  makeErrorCommunication,
} from './communicationStates';
export {
  default as makeResetCommunicationReducer,
} from './makeResetCommunicationReducer';
export { default as isFinishedCommunication } from './isFinishedCommunication';
