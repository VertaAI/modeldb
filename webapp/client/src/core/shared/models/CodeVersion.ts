import { IArtifact } from './Artifact';

export interface IArtifactCodeVersion {
  type: 'artifact';
  data: IArtifact;
}

type IIsDirtyType = 'UNKNOWN' | 'TRUE' | 'FALSE';

export interface IGitCodeVersionData {
  execPath?: string;
  remoteRepoUrl?: string;
  commitHash?: string;
  isDirty?: IIsDirtyType;
}
export interface IGitCodeVersion {
  type: 'git';
  data: IGitCodeVersionData;
}
export type ICodeVersion = IArtifactCodeVersion | IGitCodeVersion;

export interface IGitRemoteRepoUrlInfo {
  userName: string;
  repositoryInfo: { name: string; nameWithExtension: string };
}

export const parseGitRemoteRepoUrl = (
  remoteRepoUrl: string
): {
  userName: string;
  repositoryInfo: { name: string; nameWithExtension: string };
} => {
  if (remoteRepoUrl.startsWith('git@')) {
    const [_, userName, repoName] = /git@github.com:(.+)\/(.+).git/.exec(
      remoteRepoUrl
    );
    return {
      userName,
      repositoryInfo: {
        name: repoName,
        nameWithExtension: `${repoName}.git`,
      },
    };
  }
  console.assert('!!!!!');
  return {} as any;
};
