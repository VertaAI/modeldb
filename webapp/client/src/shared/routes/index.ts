import { ArgumentTypes } from 'shared/utils/types';

import { IRepository } from 'shared/models/Versioning/Repository';
import {
  CommitPointer,
  ICommit,
} from 'shared/models/Versioning/RepositoryData';
import { makeRouteFromPath, IRoute as _IRoute } from 'shared/routes/makeRoute';
import * as P from 'shared/routes/pathBuilder';
import { IURLWithFilters } from 'shared/models/Filters';
import { IURLWithResultsSorting } from 'features/highLevelSearch/url';

import {
  isRouteWithWorkspace,
  makeRouteWithWorkspace,
} from './routeWithWorkspace';
import { makeRepositoryDataWithLocationRoute } from './repositoryDataWithLocation';

export type IRoute<T, B = undefined> = _IRoute<T, B>;

export type RoutesWithWorkspaces =
  | 'projects'
  | 'datasets'
  | 'repositories'
  | 'highLevelSearch';

const routes = {
  index: makeRouteFromPath({
    getPath: () => P.makePath()(),
    allowedUserType: 'any',
  }),

  workspace: makeRouteWithWorkspace({
    getPath: () => P.makePath()(),
  }),

  highLevelSearch: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath('search')<
        { q?: string; type?: string; page?: string } & IURLWithResultsSorting
      >(),
  }),

  datasets: makeRouteWithWorkspace({
    getPath: () => P.makePath('datasets')(),
  }),
  datasetCreation: makeRouteWithWorkspace({
    getPath: () => P.makePath('datasets', 'new')(),
  }),
  datasetSummary: makeRouteWithWorkspace({
    getPath: () => P.makePath('datasets', P.param('datasetId')(), 'summary')(),
  }),
  datasetVersions: makeRouteWithWorkspace({
    getPath: () => P.makePath('datasets', P.param('datasetId')(), 'versions')(),
  }),
  datasetVersion: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'datasets',
        P.param('datasetId')(),
        'versions',
        P.param('datasetVersionId')()
      )(),
  }),
  datasetSettings: makeRouteWithWorkspace({
    getPath: () => P.makePath('datasets', P.param('datasetId')(), 'settings')(),
  }),
  compareDatasetVersions: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'datasets',
        P.param('datasetId')(),
        'versions',
        'compare',
        P.param('datasetVersionId1')(),
        P.param('datasetVersionId2')()
      )(),
  }),

  projects: makeRouteWithWorkspace({
    getPath: () => P.makePath('projects')(),
  }),
  project: makeRouteWithWorkspace({
    getPath: () => P.makePath('projects', P.param('projectId')())(),
  }),
  projectCreation: makeRouteWithWorkspace({
    getPath: () => P.makePath('projects', 'new')(),
  }),
  projectSummary: makeRouteWithWorkspace({
    getPath: () => P.makePath('projects', P.param('projectId')(), 'summary')(),
  }),
  projectSettings: makeRouteWithWorkspace({
    getPath: () => P.makePath('projects', P.param('projectId')(), 'settings')(),
  }),
  experimentRuns: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath('projects', P.param('projectId')(), 'exp-runs')<
        IURLWithFilters
      >(),
  }),
  charts: makeRouteWithWorkspace({
    getPath: () => P.makePath('projects', P.param('projectId')(), 'charts')(),
  }),
  experiments: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath('projects', P.param('projectId')(), 'experiments')(),
  }),
  experimentCreation: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath('projects', P.param('projectId')(), 'experiments', 'new')(),
  }),
  modelRecord: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'projects',
        P.param('projectId')(),
        'exp-runs',
        P.param('modelRecordId')()
      )(),
  }),
  compareModels: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'projects',
        P.param('projectId')(),
        'exp-runs',
        'compare',
        P.paramWithMod('modelIds', '*')<{
          modelIds?: string
        }>(),
      )(),
  }),

  repositories: makeRouteWithWorkspace({
    getPath: () => P.makePath('repositories')<any>(),
  }),
  createRepository: makeRouteWithWorkspace({
    getPath: () => P.makePath('repositories', 'new')(),
  }),
  repository: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'repositories',
        P.param('repositoryName')<IRepository['name']>()
      )(),
  }),
  repositoryData: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'repositories',
        P.param('repositoryName')<IRepository['name']>(),
        'data'
      )(),
  }),
  repositoryDataWithLocation: makeRepositoryDataWithLocationRoute(),
  repositoryCommitsHistory: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'repositories',
        P.param('repositoryName')<IRepository['name']>(),
        'data',
        'commits',
        P.param('commitPointerValue')<CommitPointer['value']>(),
        P.paramWithMod('locationPathname', '*')<{
          locationPathname?: string;
        }>()
      )<{ page?: string }>(),
  }),
  repositoryCommit: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'repositories',
        P.param('repositoryName')<IRepository['name']>(),
        'data',
        'commit',
        P.param('commitSha')<ICommit['sha']>()
      )(),
  }),
  repositorySettings: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'repositories',
        P.param('repositoryName')<IRepository['name']>(),
        'settings'
      )(),
  }),
  repositoryCompareChanges: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'repositories',
        P.param('repositoryName')<IRepository['name']>(),
        'data',
        'compare',
        P.param('commitPointerAValue')<CommitPointer['value']>(),
        P.param('commitPointerBValue')<CommitPointer['value']>()
      )(),
  }),
  repositoryNetworkGraph: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'repositories',
        P.param('repositoryName')<IRepository['name']>(),
        'network'
      )(),
  }),
  repositoryMergeConflicts: makeRouteWithWorkspace({
    getPath: () =>
      P.makePath(
        'repositories',
        P.param('repositoryName')<IRepository['name']>(),
        'data',
        'conflicts',
        P.param('commitPointerAValue')<CommitPointer['value']>(),
        P.param('commitPointerBValue')<CommitPointer['value']>()
      )(),
  }),
};

export const findAppRouteByPathname = (
  pathname: string,
  appRoutes: typeof routes
) => {
  const { workspace, ...routesWithoutWorkspace } = appRoutes;

  const route = Object.values(routesWithoutWorkspace).find(route_ =>
    Boolean(route_.getMatch(pathname, true))
  );

  if (route) {
    return route;
  }

  return workspace.getMatch(pathname, true) ? workspace : undefined;
};

export const checkIsAppPathnameWithoutWorkspace = (
  pathname: string,
  appRoutes: typeof routes
) => {
  const route = findAppRouteByPathname(pathname, appRoutes);
  return Boolean(route && !isRouteWithWorkspace(route));
};

export type GetRouteParams<T extends IRoute<any, any>> = ArgumentTypes<
  T['getRedirectPath']
>[0];
export type GetRouteQueryParams<
  T extends _IRoute<any, any>
> = T extends _IRoute<any, infer QueryParams> ? QueryParams : {};
export default routes;
