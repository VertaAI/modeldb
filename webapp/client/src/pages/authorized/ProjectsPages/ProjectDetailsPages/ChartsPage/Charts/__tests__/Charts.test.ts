import { ShallowWrapper } from 'enzyme';

import {
  makeShallowRenderer,
  findByDataTestAttribute,
} from 'core/shared/utils/tests/react/helpers';
import { IChartProps, Charts } from '../Charts';
import {
  projectId,
  mockExperimentRuns,
  chartConfig_default,

  // some metric missing,
  mockExperimentRuns_someMetricMissing,
  // only hyperparam
  mockExperimentRuns_onlyHyperparam,
  // only metric
  mockExperimentRuns_onlyMetric,
  // some hyperparam missing
  mockExperimentRuns_someHypMissing,
} from './mocks';

import {
  computedModelRecords,
  computedMetricHypFields,
  computedMetricHypNonNumericFields,
} from './mocks/allRowsPresent';

import {
  computedModelRecords_someDataMissing,
  computedMetricHypFields_someDataMissing,
  computedNonNumericFields_someDataMissing,
} from './mocks/someDataMissing';

import {
  chartConfig_someHypMissing,
  computedModelRecords_someHypMissing,
  computedMetricHypFields_someHypMissing,
  computedNonNumericFields_someHypMissing,
} from './mocks/someHypMissing';

import { initialCommunication } from 'core/shared/utils/redux/communication';
import { makeRouterMockProps } from 'core/shared/utils/tests/react/routeComponentProps';
import routes from 'routes';
import { userWorkspacesWithCurrentUser } from 'utils/tests/mocks/models/workspace';
import {
  chartConfig_onlyMetric,
  computedModelRecords_onlyMetrics,
  computedMetricHypFields_onlyMetrics,
  computedNonNumericFields_onlyMetric,
} from './mocks/onlyMetricsPresent';

import {
  chartConfig_onlyHyp,
  computedModelRecords_onlyHyp,
  computedMetricHypFields_onlyHyp,
  computedNonNumericFields_onlyHyp,
} from './mocks/onlyHypPresent';

const makeShallowComponent = (props: Partial<IChartProps> = {}) => {
  const routerMockProps = makeRouterMockProps(
    { projectId },
    {
      history: {
        location: {
          pathname: routes.charts.getRedirectPath({
            projectId,
            workspaceName: userWorkspacesWithCurrentUser.user.name,
          }),
        },
      },
    }
  );
  const defaultProps: IChartProps = {
    ...routerMockProps,
    sequentialChartData: [],
    loadingSequentialChartData: initialCommunication,
    lazyChartData: [],
    loadingLazyChartData: initialCommunication,
    filters: [],
    projectId,
    onResetConfigurations: jest.fn(),
    resetExperimentRunsSettings: jest.fn(),
  };
  return {
    component: makeShallowRenderer<IChartProps>(Charts, defaultProps)(props),
  };
};

const findRecord = (
  recordClass: string,
  component: ShallowWrapper<any, any, any>
) => {
  return component.findWhere(
    n => n.props().additionalContainerClassName === recordClass
  );
};

describe.skip('(charts root component)', () => {
  describe('(data: all rows with metric and hyp)', () => {
    const { component } = makeShallowComponent({
      sequentialChartData: mockExperimentRuns,
    });
    const classInstance = component.instance() as Charts;
    const state = classInstance.state;

    it('should load charts with selected default chart configuration', () => {
      const wrapper = findByDataTestAttribute('charts-root', component);
      expect(wrapper.length).toBe(1);
      expect(
        findByDataTestAttribute('charts-root', component)
      ).toMatchSnapshot();
    });

    it('should load charts with selected chart configuration', () => {
      expect(state.initialChartConfig).toMatchObject(chartConfig_default);
    });

    it('should load charts with selected chart configuration', () => {
      expect(state.chartConfigurations).toMatchObject(chartConfig_default);
    });

    it('assert chartModelRecords to be accurately computed', () => {
      expect(state.chartModelRecords).toMatchObject(computedModelRecords);
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypFields).toMatchObject(computedMetricHypFields);
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypNonNumericFields).toMatchObject(
        computedMetricHypNonNumericFields
      );
    });
  });

  describe('(data: some metric missing)', () => {
    const { component } = makeShallowComponent({
      sequentialChartData: mockExperimentRuns_someMetricMissing,
    });
    const classInstance = component.instance() as Charts;
    const state = classInstance.state;

    it('should load charts with selected default chart configuration', () => {
      const wrapper = findByDataTestAttribute('charts-root', component);
      expect(wrapper.length).toBe(1);
      expect(
        findByDataTestAttribute('charts-root', component)
      ).toMatchSnapshot();
    });

    it('should load charts with selected chart configuration', () => {
      expect(state.initialChartConfig).toMatchObject(chartConfig_default);
    });

    it('should load charts with selected chart configuration', () => {
      expect(state.chartConfigurations).toMatchObject(chartConfig_default);
    });

    it('assert chartModelRecords to be accurately computed', () => {
      expect(state.chartModelRecords).toMatchObject(
        computedModelRecords_someDataMissing
      );
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypFields).toMatchObject(
        computedMetricHypFields_someDataMissing
      );
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypNonNumericFields).toMatchObject(
        computedNonNumericFields_someDataMissing
      );
    });
  });

  describe('(data: some hyp missing)', () => {
    const { component } = makeShallowComponent({
      sequentialChartData: mockExperimentRuns_someHypMissing,
    });
    const classInstance = component.instance() as Charts;
    const state = classInstance.state;

    it('should load charts with selected default chart configuration', () => {
      const wrapper = findByDataTestAttribute('charts-root', component);
      expect(wrapper.length).toBe(1);
      expect(
        findByDataTestAttribute('charts-root', component)
      ).toMatchSnapshot();
    });

    it('should load charts with selected initial configuration', () => {
      expect(state.initialChartConfig).toMatchObject(chartConfig_default);
    });

    it('should load charts with selected chart configuration', () => {
      expect(state.chartConfigurations).toMatchObject(
        chartConfig_someHypMissing
      );
    });

    it('assert chartModelRecords to be accurately computed', () => {
      expect(state.chartModelRecords).toMatchObject(
        computedModelRecords_someHypMissing
      );
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypFields).toMatchObject(
        computedMetricHypFields_someHypMissing
      );
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypNonNumericFields).toMatchObject(
        computedNonNumericFields_someHypMissing
      );
    });
  });

  describe('(data: only metric)', () => {
    const { component } = makeShallowComponent({
      sequentialChartData: mockExperimentRuns_onlyMetric,
    });
    const classInstance = component.instance() as Charts;
    const state = classInstance.state;

    it('should load charts with selected default chart configuration', () => {
      const wrapper = findByDataTestAttribute('charts-root', component);
      expect(wrapper.length).toBe(1);
      expect(
        findByDataTestAttribute('charts-root', component)
      ).toMatchSnapshot();
    });

    it('should load charts with selected initial configuration', () => {
      expect(state.initialChartConfig).toMatchObject(chartConfig_onlyMetric);
    });

    it('should load charts with selected chart configuration', () => {
      expect(state.chartConfigurations).toMatchObject(chartConfig_onlyMetric);
    });

    it('assert chartModelRecords to be accurately computed', () => {
      expect(state.chartModelRecords).toMatchObject(
        computedModelRecords_onlyMetrics
      );
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypFields).toMatchObject(
        computedMetricHypFields_onlyMetrics
      );
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypNonNumericFields).toMatchObject(
        computedNonNumericFields_onlyMetric
      );
    });
  });

  describe('(data: only hyperparam)', () => {
    const { component } = makeShallowComponent({
      sequentialChartData: mockExperimentRuns_onlyHyperparam,
    });
    const classInstance = component.instance() as Charts;
    const state = classInstance.state;

    it('should load charts with selected default chart configuration', () => {
      const wrapper = findByDataTestAttribute('charts-root', component);
      expect(wrapper.length).toBe(1);
      expect(
        findByDataTestAttribute('charts-root', component)
      ).toMatchSnapshot();
    });

    it('should load charts with selected initial configuration', () => {
      expect(state.initialChartConfig).toMatchObject(chartConfig_onlyHyp);
    });

    it('should load charts with selected chart configuration', () => {
      expect(state.chartConfigurations).toMatchObject(chartConfig_onlyHyp);
    });

    it('assert chartModelRecords to be accurately computed', () => {
      expect(state.chartModelRecords).toMatchObject(
        computedModelRecords_onlyHyp
      );
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypFields).toMatchObject(
        computedMetricHypFields_onlyHyp
      );
    });

    it('should load charts with selected default chart configuration', () => {
      expect(state.chartMetricHypNonNumericFields).toMatchObject(
        computedNonNumericFields_onlyHyp
      );
    });
  });
});
