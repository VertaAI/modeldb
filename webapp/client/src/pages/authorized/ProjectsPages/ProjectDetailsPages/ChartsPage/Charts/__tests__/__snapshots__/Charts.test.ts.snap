// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(charts root component) (data: all rows with metric and hyp) should load charts with selected default chart configuration 1`] = `
<div
  className="root"
  data-test="charts-root"
>
  <Connect(ChartRangeFilter)
    loadingSequentialChartData={
      Object {
        "error": undefined,
        "isRequesting": false,
        "isSuccess": false,
      }
    }
    paginatedDataLength={0}
    projectId="b9edf9ac-61ad-43bf-a148-42315a25b4d4"
    resetChartConfigAtParent={[Function]}
    updateIsRangeAppliedAtParent={[Function]}
  />
  <SummaryChartManager
    genericChartData={Array []}
    metricKeys={
      Set {
        "val_acc",
      }
    }
    summaryChartConfig={
      Object {
        "selectedMetric": "data not available",
      }
    }
    updateSummaryChartConfig={[Function]}
  />
  <AggregationChartManager
    aggregationChartConfig={
      Object {
        "selectedAggregationType": "average",
        "selectedChartType": "bar-chart",
        "selectedHyperparameter": "data not available",
        "selectedMetric": "data not available",
      }
    }
    genericChartData={Array []}
    hyperparameterKeys={
      Set {
        "C",
        "max_iter",
        "solver",
      }
    }
    metricKeys={
      Set {
        "val_acc",
      }
    }
    updateAggregationChartConfig={[Function]}
  />
  <div>
    <ParallelChartManager
      data={Array []}
      metricKeysSet={
        Set {
          "val_acc",
        }
      }
      updateParallelChartConfig={[Function]}
    />
  </div>
  <div>
    <GroupedChartManager
      data={Array []}
      hyperparamKeysSet={
        Set {
          "C",
          "max_iter",
          "solver",
        }
      }
      metricKeysSet={
        Set {
          "val_acc",
        }
      }
      updateGroupedChartConfig={[Function]}
    />
  </div>
</div>
`;

exports[`(charts root component) (data: only hyperparam) should load charts with selected default chart configuration 1`] = `
<div
  className="root"
  data-test="charts-root"
>
  <Connect(ChartRangeFilter)
    loadingSequentialChartData={
      Object {
        "error": undefined,
        "isRequesting": false,
        "isSuccess": false,
      }
    }
    paginatedDataLength={0}
    projectId="b9edf9ac-61ad-43bf-a148-42315a25b4d4"
    resetChartConfigAtParent={[Function]}
    updateIsRangeAppliedAtParent={[Function]}
  />
  <SummaryChartManager
    genericChartData={Array []}
    metricKeys={Set {}}
    summaryChartConfig={
      Object {
        "selectedMetric": "data not available",
      }
    }
    updateSummaryChartConfig={[Function]}
  />
  <AggregationChartManager
    aggregationChartConfig={
      Object {
        "selectedAggregationType": "average",
        "selectedChartType": "bar-chart",
        "selectedHyperparameter": "data not available",
        "selectedMetric": "data not available",
      }
    }
    genericChartData={Array []}
    hyperparameterKeys={
      Set {
        "C",
        "max_iter",
        "solver",
      }
    }
    metricKeys={Set {}}
    updateAggregationChartConfig={[Function]}
  />
  <ChartWrapperForNoData
    canvasClassName="parallelEmptyChart"
    chartHeading="Parallel Coordinates of Hyperparameters and Metrics"
    message="No parallel Coordinates for available data"
  />
  <ChartWrapperForNoData
    canvasClassName="groupedEmptyChart"
    chartHeading="Grouped Metric Chart, Aggregated by Hyperparameters"
    message="No grouping results for available data"
  />
</div>
`;

exports[`(charts root component) (data: only metric) should load charts with selected default chart configuration 1`] = `
<div
  className="root"
  data-test="charts-root"
>
  <Connect(ChartRangeFilter)
    loadingSequentialChartData={
      Object {
        "error": undefined,
        "isRequesting": false,
        "isSuccess": false,
      }
    }
    paginatedDataLength={0}
    projectId="b9edf9ac-61ad-43bf-a148-42315a25b4d4"
    resetChartConfigAtParent={[Function]}
    updateIsRangeAppliedAtParent={[Function]}
  />
  <SummaryChartManager
    genericChartData={Array []}
    metricKeys={
      Set {
        "val_acc",
      }
    }
    summaryChartConfig={
      Object {
        "selectedMetric": "data not available",
      }
    }
    updateSummaryChartConfig={[Function]}
  />
  <AggregationChartManager
    aggregationChartConfig={
      Object {
        "selectedAggregationType": "average",
        "selectedChartType": "bar-chart",
        "selectedHyperparameter": "data not available",
        "selectedMetric": "data not available",
      }
    }
    genericChartData={Array []}
    hyperparameterKeys={Set {}}
    metricKeys={
      Set {
        "val_acc",
      }
    }
    updateAggregationChartConfig={[Function]}
  />
  <div>
    <ParallelChartManager
      data={Array []}
      initialConfiguration={
        Object {
          "selectedPanelElements": Array [],
        }
      }
      metricKeysSet={
        Set {
          "val_acc",
        }
      }
      updateParallelChartConfig={[Function]}
    />
  </div>
  <ChartWrapperForNoData
    canvasClassName="groupedEmptyChart"
    chartHeading="Grouped Metric Chart, Aggregated by Hyperparameters"
    message="No grouping results for available data"
  />
</div>
`;

exports[`(charts root component) (data: some hyp missing) should load charts with selected default chart configuration 1`] = `
<div
  className="root"
  data-test="charts-root"
>
  <Connect(ChartRangeFilter)
    loadingSequentialChartData={
      Object {
        "error": undefined,
        "isRequesting": false,
        "isSuccess": false,
      }
    }
    paginatedDataLength={0}
    projectId="b9edf9ac-61ad-43bf-a148-42315a25b4d4"
    resetChartConfigAtParent={[Function]}
    updateIsRangeAppliedAtParent={[Function]}
  />
  <SummaryChartManager
    genericChartData={Array []}
    metricKeys={
      Set {
        "val_acc",
      }
    }
    summaryChartConfig={
      Object {
        "selectedMetric": "data not available",
      }
    }
    updateSummaryChartConfig={[Function]}
  />
  <AggregationChartManager
    aggregationChartConfig={
      Object {
        "selectedAggregationType": "average",
        "selectedChartType": "bar-chart",
        "selectedHyperparameter": "data not available",
        "selectedMetric": "data not available",
      }
    }
    genericChartData={Array []}
    hyperparameterKeys={
      Set {
        "C",
        "max_iter",
        "solver",
      }
    }
    metricKeys={
      Set {
        "val_acc",
      }
    }
    updateAggregationChartConfig={[Function]}
  />
  <div>
    <ParallelChartManager
      data={Array []}
      initialConfiguration={
        Object {
          "selectedPanelElements": Array [],
        }
      }
      metricKeysSet={
        Set {
          "val_acc",
        }
      }
      updateParallelChartConfig={[Function]}
    />
  </div>
  <div>
    <GroupedChartManager
      data={Array []}
      hyperparamKeysSet={
        Set {
          "C",
          "max_iter",
          "solver",
        }
      }
      initialConfiguration={
        Object {
          "selectedChartType": "multi-line-chart",
          "selectedPanelElements": Array [],
          "selectedYAxisType": "linear",
        }
      }
      metricKeysSet={
        Set {
          "val_acc",
        }
      }
      updateGroupedChartConfig={[Function]}
    />
  </div>
</div>
`;

exports[`(charts root component) (data: some metric missing) should load charts with selected default chart configuration 1`] = `
<div
  className="root"
  data-test="charts-root"
>
  <Connect(ChartRangeFilter)
    loadingSequentialChartData={
      Object {
        "error": undefined,
        "isRequesting": false,
        "isSuccess": false,
      }
    }
    paginatedDataLength={0}
    projectId="b9edf9ac-61ad-43bf-a148-42315a25b4d4"
    resetChartConfigAtParent={[Function]}
    updateIsRangeAppliedAtParent={[Function]}
  />
  <SummaryChartManager
    genericChartData={Array []}
    metricKeys={
      Set {
        "val_acc",
      }
    }
    summaryChartConfig={
      Object {
        "selectedMetric": "data not available",
      }
    }
    updateSummaryChartConfig={[Function]}
  />
  <AggregationChartManager
    aggregationChartConfig={
      Object {
        "selectedAggregationType": "average",
        "selectedChartType": "bar-chart",
        "selectedHyperparameter": "data not available",
        "selectedMetric": "data not available",
      }
    }
    genericChartData={Array []}
    hyperparameterKeys={
      Set {
        "C",
        "max_iter",
        "solver",
      }
    }
    metricKeys={
      Set {
        "val_acc",
      }
    }
    updateAggregationChartConfig={[Function]}
  />
  <div>
    <ParallelChartManager
      data={Array []}
      initialConfiguration={
        Object {
          "selectedPanelElements": Array [],
        }
      }
      metricKeysSet={
        Set {
          "val_acc",
        }
      }
      updateParallelChartConfig={[Function]}
    />
  </div>
  <div>
    <GroupedChartManager
      data={Array []}
      hyperparamKeysSet={
        Set {
          "C",
          "max_iter",
          "solver",
        }
      }
      initialConfiguration={
        Object {
          "selectedChartType": "multi-line-chart",
          "selectedPanelElements": Array [],
          "selectedYAxisType": "linear",
        }
      }
      metricKeysSet={
        Set {
          "val_acc",
        }
      }
      updateGroupedChartConfig={[Function]}
    />
  </div>
</div>
`;
